openhcl/page_pool_alloc/src/lib.rs

```diff
@@ -198,13 +511,12 @@ impl Drop for PagePoolHandle {
 /// [`PagePoolAllocatorSpawner::allocator`].
 ///
 /// This struct is considered the "owner" of the pool allowing for save/restore.
-///
-// TODO SNP: Implement save restore. This means additionally having some sort of
-// restore_alloc method that maps to an existing allocation.
 #[derive(Inspect)]
 pub struct PagePool {
     #[inspect(flatten)]
     inner: Arc<Mutex<PagePoolInner>>,
+    #[inspect(iter_by_index)]
+    ranges: Vec<MemoryRangeWithNode>,
```

Comment:
Do you really need the vnodes?

Reply:
Eh not really right now. In the future maybe we want numa aligned allocations but i'll leave that as a future problem